subprojects/gdk_rust/gdk_common/src/be/mod.rs

```diff
@@ -32,34 +32,53 @@ pub struct UTXOInfo {
 }
 
 impl UTXOInfo {
-    pub fn new(
-        asset: String,
+    pub fn new_bitcoin(
         value: u64,
         script: BEScript,
         height: Option<u32>,
         path: DerivationPath,
     ) -> Self {
         UTXOInfo {
-            asset,
+            asset: "btc".to_string(),
             value,
             script,
             height,
             path,
         }
     }
+
+    pub fn new_elements(
+        asset: elements::issuance::AssetId,
+        value: u64,
+        script: BEScript,
+        height: Option<u32>,
+        path: DerivationPath,
+    ) -> Self {
+        UTXOInfo {
+            asset: asset.to_hex(),
+            value,
+            script,
+            height,
+            path,
+        }
+    }
+
+    pub fn asset_id(&self) -> Option<elements::issuance::AssetId> {
+        self.asset.parse().ok()
```

Comment:
Maybe worth explicitly checking for `btc` and returning a `None` instead of trying and failing the hex conversion?

Also, for the non-`btc` case, maybe better to fail loudly if parsing fails? (I think that panicking could be appropriate for this)

Reply:
Yes, that's better, added in 79afbcb

Reply:
Looks good!